packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -139,7 +139,46 @@ function useConfigLoader(
     };
   }
 
-  const configReader = ConfigReader.fromConfigs(config.value ?? []);
+  let configReader = ConfigReader.fromConfigs(config.value ?? []);
+
+  const resolveRelativeUrl = (relativeUrl: string) =>
+    new URL(relativeUrl, document.location.origin).href;
+
+  const getRelativeUrl = (fullUrl: string) => {
+    const url = new URL(fullUrl);
+    return fullUrl.replace(url.origin, '');
+  };
+
+  /**
+   * We only want to override the URLs with the document origin when the URLs match
+   *  and are defined. We use getOptionalString here to not throw when the app.baseUrl
+   *  and backend.baseUrl are not defined. If they are defined but not well formatted URLs
+   *  the above getRelativeUrl() method will throw.
+   */
+  if (
+    configReader.getOptionalString('app.baseUrl') &&
+    configReader.getOptionalString('backend.baseUrl') &&
+    configReader.getOptionalString('app.baseUrl') ===
+      configReader.getOptionalString('backend.baseUrl')
```

Comment:
Let's read these out separately and store in a variable for later calculations 😅. It's fine to `.getString` btw, we do that later anyway. I believe the equality check should also just be for the origin of the URLs, it doesn't matter if the paths are different.